source/documentation/using_the_api/register_components.md

```diff
@@ -1,18 +1,30 @@
-## The relationship between entries, items and records
+## The components of a register
 
-Registers are made up of:
+Registers are primarily composed of:
 
 * entries
 * items
 * records
 
+### Entries
+
 Entries contain metadata about what and when data has changed. 
 
-Each entry in a register has a relationship with one item, which contains data. An item is connected to an entry with an item hash.   
+Entries are made up of the fields `entry-number`, `index-entry-number`, `entry-timestamp`, `item-hash` and `key`. 
+
+The `entry-number` is unique and defines an entry’s position within the ordered list of a register. The `index-entry-number` is unique and defines an entry’s position within the ordered list of an index. For any given entry in a register, the `entry-number` and `index-entry-number` are always identical. 
+
+The `entry-timestamp` is the time when an entry was introduced to a given register, but it is no guarantee of the order of entries in a register. The `entry-number` is.
```

Comment:
I think `key` should be described as well. And probably explain it has the same value as the identifier (primary key) in the item.

Reply:
Yes, this is actually an oversight (not describing `key`). Would this be a good opportunity to introduce foreign keys, juxtaposed against primary keys... something like this:

The `key` is a numeric or alphanumeric string. The primary key always has the same name as the name of the register. For example, every entry in the `local-authority-eng` register has a field called `local-authority-eng`. [Foreign keys, which are not always present, link to other registers](link-to-content-about-foreign-keys).

Reply:
> The key is a numeric or alphanumeric string

"The key is an alphanumeric string" is better given that even if it only contains numbers it fits the definition. The point is that you must not treat it as a number ever.

Reply:
I like how you introduce the foreign key concept. If possible, it would be good to equal "primary key" to "unique identifier". Mainly because ID is a concept more popular in other HTTP APIs and at some point I would like to use "id" as the field name.

Reply:
I was also wondering about "alphanumeric": the definitions I've found seem to suggest that "alphanumeric" means "a mix of letters and numbers" whereas "numeric" means "just numbers". I'm inclined to agree with you, but it's an interesting question.

On the second point, I'll rework it such that it implies an equivalence with primary key and unique identifier.